### ISSUE TITLE
google.com: resolve call failed: 'google.com' does not have any RR of the requested type

### ISSUE THREAD

**Issue by tpgxyz** [systemd/systemd#18917]
**systemd version the issue has been seen with**
 > 247

**Used distribution**
 > OpenMandriva 4.2

**Linux kernel version used** (`uname -a`)
 > `Linux tpg-latitude5490 5.10.14-desktop-1omv4002 #1 SMP PREEMPT Mon Feb 8 18:13:34 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
`

**CPU architecture issue was seen on**
 > x86_64

**Expected behaviour you didn't see**
 > DNS resolve works

**Unexpected behaviour you saw**
 > DNS resolve does not work

**Steps to reproduce the problem**
 > make syre systemd-resolved.service is enabled and running
ping google.com -> no resolve
resolvectl query google.com -> no resolve

**Additional program output to the terminal or log subsystem illustrating the issue**

```
[root@tpg-latitude5490 tpg]# resolvectl query google.com
google.com: resolve call failed: 'google.com' does not have any RR of the requested type

```
Output from SYSTEMD_LOG_LEVEL=debug  /lib/systemd/systemd-resolved

```
[root@tpg-latitude5490 tpg]# SYSTEMD_LOG_LEVEL=debug  /lib/systemd/systemd-resolved 
Positive Trust Anchors:
. IN DS 20326 8 2 e06d44b80b8f1d39a95c0b0d7c65d08458e880409bbc683457104237c7f8ec8d
Negative trust anchors: 10.in-addr.arpa 16.172.in-addr.arpa 17.172.in-addr.arpa 18.172.in-addr.arpa 19.172.in-addr.arpa 20.172.in-addr.arpa 21.172.in-addr.arpa 22.172.in-addr.arpa 23.172.in-addr.arpa 24.172.in-addr.arpa 25.172.in-addr.arpa 26.172.in-addr.arpa 27.172.in-addr.arpa 28.172.in-addr.arpa 29.172.in-addr.arpa 30.172.in-addr.arpa 31.172.in-addr.arpa 168.192.in-addr.arpa d.f.ip6.arpa corp home internal intranet lan local private test
Using system hostname 'tpg-latitude5490'.
New scope on link *, protocol dns, family *
Found new link 8/wlan0
Found new link 2/enp0s31f6
Found new link 1/lo
New scope on link wlan0, protocol llmnr, family AF_INET6
Transaction 17483 for <tpg-latitude5490 IN ANY> scope llmnr on wlan0/INET6.
Delaying llmnr transaction for 61151us.
New scope on link wlan0, protocol dns, family *
New scope on link wlan0, protocol llmnr, family AF_INET
Transaction 26499 for <tpg-latitude5490 IN ANY> scope llmnr on wlan0/INET.
Delaying llmnr transaction for 32037us.
Bus bus-api-resolve: changing state UNSET → OPENING
sd-bus: starting bus bus-api-resolve by connecting to /run/dbus/system_bus_socket...
Bus bus-api-resolve: changing state OPENING → AUTHENTICATING
Registering bus object implementation for path=/org/freedesktop/resolve1 iface=org.freedesktop.resolve1.Manager
Registering bus object implementation for path=/org/freedesktop/resolve1/link iface=org.freedesktop.resolve1.Link
Registering bus object implementation for path=/org/freedesktop/resolve1/dnssd iface=org.freedesktop.resolve1.DnssdService
Registering bus object implementation for path=/org/freedesktop/LogControl1 iface=org.freedesktop.LogControl1
Creating stub listener using UDP/TCP.
Bus bus-api-resolve: changing state AUTHENTICATING → HELLO
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=Hello cookie=1 reply_cookie=0 signature=n/a error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=RequestName cookie=2 reply_cookie=0 signature=su error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=AddMatch cookie=3 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.181 path=n/a interface=n/a member=n/a cookie=4294967295 reply_cookie=1 signature=s error-name=n/a error-message=n/a
Bus bus-api-resolve: changing state HELLO → RUNNING
Got message type=signal sender=org.freedesktop.DBus.Local destination=n/a path=/org/freedesktop/DBus/Local interface=org.freedesktop.DBus.Local member=Connected cookie=4294967295 reply_cookie=0 signature=n/a error-name=n/a error-message=n/a
Got message type=signal sender=org.freedesktop.DBus destination=:1.181 path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=NameAcquired cookie=4294967295 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=signal sender=org.freedesktop.DBus destination=:1.181 path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=NameAcquired cookie=4294967295 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.181 path=n/a interface=n/a member=n/a cookie=4294967295 reply_cookie=2 signature=u error-name=n/a error-message=n/a
Successfully acquired requested service name.
Got message type=method_return sender=org.freedesktop.DBus destination=:1.181 path=n/a interface=n/a member=n/a cookie=4294967295 reply_cookie=3 signature= error-name=n/a error-message=n/a
Match type='signal',sender='org.freedesktop.login1',path='/org/freedesktop/login1',interface='org.freedesktop.login1.Manager',member='PrepareForSleep' successfully installed.
Got message type=method_call sender=:1.168 destination=org.freedesktop.resolve1 path=/org/freedesktop/resolve1 interface=org.freedesktop.resolve1.Manager member=SetLinkDomains cookie=1204 reply_cookie=0 signature=ia(sb) error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=GetConnectionUnixUser cookie=4 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.181 path=n/a interface=n/a member=n/a cookie=4294967295 reply_cookie=4 signature=u error-name=n/a error-message=n/a
Sent message type=method_return sender=n/a destination=:1.168 path=n/a interface=n/a member=n/a cookie=5 reply_cookie=1204 signature=n/a error-name=n/a error-message=n/a
Got message type=method_call sender=:1.168 destination=org.freedesktop.resolve1 path=/org/freedesktop/resolve1 interface=org.freedesktop.resolve1.Manager member=SetLinkDefaultRoute cookie=1205 reply_cookie=0 signature=ib error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=GetConnectionUnixUser cookie=6 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.181 path=n/a interface=n/a member=n/a cookie=4294967295 reply_cookie=6 signature=u error-name=n/a error-message=n/a
Sent message type=method_return sender=n/a destination=:1.168 path=n/a interface=n/a member=n/a cookie=7 reply_cookie=1205 signature=n/a error-name=n/a error-message=n/a
Got message type=method_call sender=:1.168 destination=org.freedesktop.resolve1 path=/org/freedesktop/resolve1 interface=org.freedesktop.resolve1.Manager member=SetLinkMulticastDNS cookie=1206 reply_cookie=0 signature=is error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=GetConnectionUnixUser cookie=8 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.181 path=n/a interface=n/a member=n/a cookie=4294967295 reply_cookie=8 signature=u error-name=n/a error-message=n/a
Sent message type=method_return sender=n/a destination=:1.168 path=n/a interface=n/a member=n/a cookie=9 reply_cookie=1206 signature=n/a error-name=n/a error-message=n/a
Got message type=method_call sender=:1.168 destination=org.freedesktop.resolve1 path=/org/freedesktop/resolve1 interface=org.freedesktop.resolve1.Manager member=SetLinkLLMNR cookie=1207 reply_cookie=0 signature=is error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=GetConnectionUnixUser cookie=10 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.181 path=n/a interface=n/a member=n/a cookie=4294967295 reply_cookie=10 signature=u error-name=n/a error-message=n/a
Sent message type=method_return sender=n/a destination=:1.168 path=n/a interface=n/a member=n/a cookie=11 reply_cookie=1207 signature=n/a error-name=n/a error-message=n/a
Got message type=method_call sender=:1.168 destination=org.freedesktop.resolve1 path=/org/freedesktop/resolve1 interface=org.freedesktop.resolve1.Manager member=SetLinkDNS cookie=1208 reply_cookie=0 signature=ia(iay) error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=GetConnectionUnixUser cookie=12 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.181 path=n/a interface=n/a member=n/a cookie=4294967295 reply_cookie=12 signature=u error-name=n/a error-message=n/a
Sent message type=signal sender=n/a destination=n/a path=/org/freedesktop/resolve1 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=13 reply_cookie=0 signature=sa{sv}as error-name=n/a error-message=n/a
Sent message type=method_return sender=n/a destination=:1.168 path=n/a interface=n/a member=n/a cookie=14 reply_cookie=1208 signature=n/a error-name=n/a error-message=n/a
Timeout reached on transaction 26499.
Retrying transaction 26499.
Transaction 26499 for <tpg-latitude5490 IN ANY> scope llmnr on wlan0/INET.
Sending query packet with id 26499 on interface 8/AF_INET of size 34.
Timeout reached on transaction 17483.
Retrying transaction 17483.
Transaction 17483 for <tpg-latitude5490 IN ANY> scope llmnr on wlan0/INET6.
Sending query packet with id 17483 on interface 8/AF_INET6 of size 34.
Timeout reached on transaction 26499.
Retrying transaction 26499.
Transaction 26499 for <tpg-latitude5490 IN ANY> scope llmnr on wlan0/INET.
Sending query packet with id 26499 on interface 8/AF_INET of size 34.
Timeout reached on transaction 17483.
Retrying transaction 17483.
Transaction 17483 for <tpg-latitude5490 IN ANY> scope llmnr on wlan0/INET6.
Sending query packet with id 17483 on interface 8/AF_INET6 of size 34.
Timeout reached on transaction 26499.
Retrying transaction 26499.
Transaction 26499 for <tpg-latitude5490 IN ANY> scope llmnr on wlan0/INET.
Sending query packet with id 26499 on interface 8/AF_INET of size 34.
Timeout reached on transaction 17483.
Retrying transaction 17483.
Transaction 17483 for <tpg-latitude5490 IN ANY> scope llmnr on wlan0/INET6.
Sending query packet with id 17483 on interface 8/AF_INET6 of size 34.
Timeout reached on transaction 26499.
Retrying transaction 26499.
Transaction 26499 for <tpg-latitude5490 IN ANY> on scope llmnr on wlan0/INET now complete with <not-found> from none (unsigned).
Record tpg-latitude5490 IN A 192.168.43.164 successfully probed.
Freeing transaction 26499.
Timeout reached on transaction 17483.
Retrying transaction 17483.
Transaction 17483 for <tpg-latitude5490 IN ANY> on scope llmnr on wlan0/INET6 now complete with <not-found> from none (unsigned).
Record tpg-latitude5490 IN AAAA 2a00:f41:1c2e:6ffc:f5f7:3e95:cc9:5625 successfully probed.
Record tpg-latitude5490 IN AAAA fe80::3dc3:5d26:feab:2d6c successfully probed.
Freeing transaction 17483.
Got message type=method_call sender=:1.182 destination=org.freedesktop.resolve1 path=/org/freedesktop/resolve1 interface=org.freedesktop.resolve1.Manager member=ResolveHostname cookie=2 reply_cookie=0 signature=isit error-name=n/a error-message=n/a
idn2_lookup_u8: google.com → google.com
Looking up RR for google.com IN A.
Looking up RR for google.com IN AAAA.
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=AddMatch cookie=15 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=GetNameOwner cookie=16 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.181 path=n/a interface=n/a member=n/a cookie=4294967295 reply_cookie=16 signature=s error-name=n/a error-message=n/a
Switching to DNS server 192.168.43.69 for interface wlan0.
Cache miss for google.com IN A
Transaction 56186 for <google.com IN A> scope dns on wlan0/*.
Using feature level UDP+EDNS0 for transaction 56186.
Using DNS server 192.168.43.69 for transaction 56186.
Sending query packet with id 56186 of size 39.
Cache miss for google.com IN AAAA
Transaction 62663 for <google.com IN AAAA> scope dns on wlan0/*.
Using feature level UDP+EDNS0 for transaction 62663.
Using DNS server 192.168.43.69 for transaction 62663.
Sending query packet with id 62663 of size 39.
Got message type=method_return sender=org.freedesktop.DBus destination=:1.181 path=n/a interface=n/a member=n/a cookie=4294967295 reply_cookie=15 signature= error-name=n/a error-message=n/a
Match type='signal',sender='org.freedesktop.DBus',path='/org/freedesktop/DBus',interface='org.freedesktop.DBus',member='NameOwnerChanged',arg0=':1.182' successfully installed.
Processing incoming packet on transaction 56186 (rcode=SUCCESS).
More resource records declared in packet than included, suppressing OPT.
Verified we get a response at feature level UDP from DNS server 192.168.43.69.
Server doesn't support EDNS(0) properly, downgrading feature level...
Using degraded feature set UDP instead of UDP+EDNS0 for DNS server 192.168.43.69.
Server feature level is now lower than when we began our transaction. Restarting with new ID.
Transaction 56186 is now 15625.
Cache miss for google.com IN A
Transaction 15625 for <google.com IN A> scope dns on wlan0/*.
Using feature level UDP for transaction 15625.
Sending query packet with id 15625 of size 28.
Processing incoming packet on transaction 62663 (rcode=SUCCESS).
More resource records declared in packet than included, suppressing OPT.
Server feature level is now lower than when we began our transaction. Restarting with new ID.
Transaction 62663 is now 41596.
Cache miss for google.com IN AAAA
Transaction 41596 for <google.com IN AAAA> scope dns on wlan0/*.
Using feature level UDP for transaction 41596.
Sending query packet with id 41596 of size 28.
Processing incoming packet on transaction 15625 (rcode=SUCCESS).
More resource records declared in packet than included, suppressing OPT.
Not caching negative entry without a SOA record: google.com IN A
Transaction 15625 for <google.com IN A> on scope dns on wlan0/* now complete with <success> from network (unsigned).
Processing incoming packet on transaction 41596 (rcode=SUCCESS).
More resource records declared in packet than included, suppressing OPT.
Not caching negative entry without a SOA record: google.com IN AAAA
Transaction 41596 for <google.com IN AAAA> on scope dns on wlan0/* now complete with <success> from network (unsigned).
Freeing transaction 15625.
Sent message type=error sender=n/a destination=:1.182 path=n/a interface=n/a member=n/a cookie=17 reply_cookie=2 signature=s error-name=org.freedesktop.resolve1.NoSuchRR error-message='google.com' does not have any RR of the requested type
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=RemoveMatch cookie=18 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Freeing transaction 41596.
^CBus bus-api-resolve: changing state RUNNING → CLOSED
Removing scope on link wlan0, protocol dns, family *
Removing scope on link wlan0, protocol llmnr, family AF_INET
Removing scope on link wlan0, protocol llmnr, family AF_INET6
Removing scope on link *, protocol dns, family *
```

```
[root@tpg-latitude5490 tpg]# resolvectl status
Global
           Protocols: +LLMNR +mDNS -DNSOverTLS DNSSEC=no/unsupported
    resolv.conf mode: uplink                                        
Fallback DNS Servers: 208.67.222.222 208.67.220.220                 

Link 2 (enp0s31f6)
Current Scopes: none                                                        
     Protocols: -DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported
   DNS Servers: 208.67.220.220                                              

Link 8 (wlan0)
Current Scopes: DNS LLMNR/IPv4 LLMNR/IPv6                                   
     Protocols: +DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported
   DNS Servers: 192.168.43.69 2a00:f41:1c2e:6ffc::ad  
```

/etc/nsswitch.conf
`hosts:          resolve files myhostname dns nis wins mdns4_minimal`

/etc/resolv.conf is symlinked to  /run/systemd/resolve/stub-resolv.conf - when i change symlink to /run/systemd/resolve/resolv.conf  a name resolution starts to work.

```
[root@tpg-latitude5490 tpg]# cat /run/systemd/resolve/resolv.conf 
nameserver 192.168.43.69
nameserver 2a00:f41:1c2e:6ffc::ad
nameserver 208.67.220.220
search .

[root@tpg-latitude5490 tpg]# cat /run/systemd/resolve/stub-resolv.conf 
nameserver 127.0.0.53
options edns0 trust-ad
search .

```

```
[root@tpg-latitude5490 tpg]# netstat -nlutp
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:5355            0.0.0.0:*               LISTEN      703/systemd-resolve 
tcp        0      0 0.0.0.0:111             0.0.0.0:*               LISTEN      1/systemd           
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      703/systemd-resolve 
tcp        0      0 0.0.0.0:631             0.0.0.0:*               LISTEN      837/cupsd           
tcp        0      0 127.0.0.1:44835         0.0.0.0:*               LISTEN      1472/falkon         
tcp        0      0 127.0.0.1:2947          0.0.0.0:*               LISTEN      1/systemd           
tcp6       0      0 :::5355                 :::*                    LISTEN      703/systemd-resolve 
tcp6       0      0 :::111                  :::*                    LISTEN      1/systemd           
tcp6       0      0 :::1716                 :::*                    LISTEN      1168/kdeconnectd    
tcp6       0      0 :::22                   :::*                    LISTEN      1/systemd           
tcp6       0      0 :::631                  :::*                    LISTEN      837/cupsd           
tcp6       0      0 ::1:2947                :::*                    LISTEN      1/systemd           
udp        0      0 127.0.0.53:53           0.0.0.0:*                           703/systemd-resolve 
udp        0      0 0.0.0.0:111             0.0.0.0:*                           1/systemd           
udp        0      0 0.0.0.0:5355            0.0.0.0:*                           703/systemd-resolve 
udp6       0      0 :::111                  :::*                                1/systemd           
udp6       0      0 :::1716                 :::*                                1168/kdeconnectd    
udp6       0      0 :::5355                 :::*                                703/systemd-resolve
```

**Comment by poettering** [comment]
These lines are interesting:

> More resource records declared in packet than included, suppressing OPT.

Thee suggests your DNS packet are truncated somehow? i.e. as if the replies we get from your DNS server are cut in the middle so that the address data included is dropped What kind of DNS server is this?

Any chance you can provide a pcap file (wireshark) of the relevant DNS traffic?

**Comment by poettering** [comment]
That message was added in 18674159ebbf622a9e6e5a45cc36b38f74dae315 which is post v247. This indicates that your distro *heavily* backported patches. I wonder if everything went right with that.

**Comment by tpgxyz** [comment]
@poettering  yes I'm using up to date snapshot from systemd/systemd-stable

I have never observed such issue. Today I was using my Samsung smartphone as a WiFi router for my laptop.

**Comment by poettering** [comment]
192.168.43.69, is that your cell phone? or the DNS server the cell is connected to?

no idea how tethering on android actually works. does it implement a simple DNS proxy? anyone knows?

but anyway, please provide the pcap of the DNS traffic to that server, i.e. `tcpdump -n -w github18917.pcap udp port 53`, then recreate the issue and attach the generated file `github18917.pcap` here. The more isolated you can make the DNS traffic the better. And please consider doing a "resolvectl flush-caches" immediately before starting the capture, so that the requests are always served from the network

**Comment by tpgxyz** [comment]
@poettering Bero from OpenMandrriva after updating to latest system/systemd-stable started to experience DNS issues. After updating to 248-rc2 these were gone.

Anyways I'll prepare the tcpdump.

**Comment by tpgxyz** [comment]
@poettering here you go a tcpdump
[github18917.pcap.zip](https://github.com/systemd/systemd/files/6099938/github18917.pcap.zip)

**Comment by keszybz** [comment]
> Hmm, so the pcap looks fine. I wonder if the backport is correct then. @keszybz?

The packport seemed rather straightforward. Most likely some other patch that wasn't yet backported is relevant.

**Comment by tpgxyz** [comment]
@keszybz @poettering any news on this ? systemd-stable seems to be broken.

**Comment by akiirui** [comment]
Same issue on Arch Linux `testing/systemd 247.4-1`

**Comment by andreaskem** [comment]
I also encountered this issue on Arch Linux immediately after upgrading to 274.4. A downgrade to .3 made everything work again. Some domains could still be resolved properly but others were completely broken. It occurred to me that working domains might not have had an IPv6 address (e.g., arstechnica.com).
```
andreas@e485-arch ~ $ resolvectl
Global
           Protocols: +LLMNR +mDNS -DNSOverTLS DNSSEC=no/unsupported                                
    resolv.conf mode: stub                                                                          
Fallback DNS Servers: 1.1.1.1 9.9.9.10 8.8.8.8 2606:4700:4700::1111 2620:fe::10 2001:4860:4860::8888

Link 4 (wlan0)
    Current Scopes: DNS LLMNR/IPv4 LLMNR/IPv6                                   
         Protocols: +DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported
Current DNS Server: 192.168.178.1                                               
       DNS Servers: 192.168.178.1 fd00::7eff:4dff:fec4:e0ec
```

**Comment by batGnat** [comment]
Also have same issue on Arch Linux testing/systemd 247.4-1. 
I changed nsswitch.conf to prioritise DNS over RESOLVE for temp solution.

**Comment by morgoth6** [comment]
Reverting https://github.com/systemd/systemd-stable/commit/18674159ebbf622a9e6e5a45cc36b38f74dae315 helped here. First query still fails, but next one seems to be OK.

**Comment by batGnat** [comment]
This seems to be fixed on Arch Linux 247.4-2
```
batgnat@batlenovo~> resolvectl query www.google.com
www.google.com: 2404:6800:4006:810::2004       -- link: wlan0
                142.250.66.228                 -- link: wlan0

-- Information acquired via protocol DNS in 1.2ms.
-- Data is authenticated: no
batgnat@batlenovo~> resolvectl status
Global
           Protocols: +LLMNR +mDNS -DNSOverTLS DNSSEC=no/unsupported                                
    resolv.conf mode: foreign                                                                       
  Current DNS Server: 1.1.1.1                                                                       
         DNS Servers: 1.1.1.1 1.0.0.1 2606:4700:4700::1111 2606:4700:4700::1001 10.0.0.1            
Fallback DNS Servers: 1.1.1.1 9.9.9.10 8.8.8.8 2606:4700:4700::1111 2620:fe::10 2001:4860:4860::8888

Link 2 (enp3s0)
Current Scopes: none                                                        
     Protocols: -DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported

Link 3 (wlan0)
Current Scopes: LLMNR/IPv4 LLMNR/IPv6                                       
     Protocols: -DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported
```

**Comment by eworm-de** [comment]
The Arch Linux package `systemd` `247.4-2` reverts 18674159ebbf622a9e6e5a45cc36b38f74dae315.

**Comment by tpgxyz** [comment]
Yes i can confirm that reverting https://github.com/systemd/systemd/commit/18674159ebbf622a9e6e5a45cc36b38f74dae315 does help, but that is a workaround, not a *real* solution.

**Comment by sandy-8925** [comment]
Well it sounds like [1867415](https://github.com/systemd/systemd/commit/18674159ebbf622a9e6e5a45cc36b38f74dae315) is a bug that was introduced in the code.

**Comment by tpgxyz** [comment]
New snapshot of systemd-stable plus revert of commit 1867415 finally make it works.

```
[root@tpg-latitude5490 tpg]# resolvectl query google.com
google.com: 2a00:1450:401b:808::200e           -- link: wlan0
            216.58.209.14                      -- link: wlan0

-- Information acquired via protocol DNS in 29.3ms.
-- Data is authenticated: no

```

**Comment by fbuihuu** [comment]
@keszybz do you plan to fix the stable trees (either by reverting 1867415 or by adding the other missing relevant patches) ?

**Comment by poettering** [comment]
@tpgxyz @fbuihuu any chance you can gdb through this? I can't reproduce this here. i.e. set the a breakpoint on the "More resource records declared in packet than included, suppressing OPT." log line, then run this, so the issue reproduces.generate a "bt full", plus an explicit dump of the `*p` structure. It would also be great to extract `p->size` bytes from the `p->_data` pointer (use `dump binary memory p->_data ((uint8_t*) p->data + p->size)` or something like that)

(and no backports please, please use git master)

**Comment by fbuihuu** [comment]
> @tpgxyz @fbuihuu any chance you can gdb through this?

I can't reproduce it myself, I was just asking because several users running Tumbleweed (v246.11) reported this issue too.

> (and no backports please, please use git master)

Using "main" wouldn't help since according to https://github.com/systemd/systemd/issues/18917#issuecomment-792487807, the issue is gone with v248-rc2.

**Comment by tpgxyz** [comment]
@fbuihuu that comment got superseded by 
https://github.com/systemd/systemd/issues/18917#issuecomment-795625459

**Comment by keszybz** [comment]
> @keszybz do you plan to fix the stable trees (either by reverting [1867415](https://github.com/systemd/systemd/commit/18674159ebbf622a9e6e5a45cc36b38f74dae315) or by adding the other missing relevant patches) ?

Reverted in v246.13 and v247.6.

**Comment by poettering** [comment]
Only way we can address this, if the data requested in https://github.com/systemd/systemd/issues/18917#issuecomment-804722158 is provided.

**Comment by tpgxyz** [comment]
> @tpgxyz @fbuihuu any chance you can gdb through this?

@poettering Can you please help here. Below are the steps i took:
1. gdb -p $PID_OF_RESOLVE
2. break "More resource records declared in packet than included, suppressing OPT."
3. ping google.com
4. bt full <- empty
5. dump binary memory p->_data ((uint8_t*) p->_data + p->size) <- it says its empty and i should use file

**Comment by poettering** [comment]
Your step 2 must be changed to set a break point on a source line, the source line that generates that log message. i.e. that log_debug(…) line in dns_packet_extract_answer(). Must be something like `b resolved-dns-packet.c:2281`. (That's the line here in my hacked version, it's probably a different line for your patched version).

Between step 2 and three you need to do "continue" (aka "c") to continue running it.

Can you provide the exact output of that "dump binary" error message?

What does `print *p` say when the break point hits? Please paste the output here

**Comment by poettering** [comment]
I'd be curious if #19108 changes anything about this issue btw. I mean, I doubt it's a fix, but I really don't grok the issue, hence I'd really like to know if that changes anything.

**Comment by keszybz** [comment]
Based on various testing done by contributors, it seems that this either does not occur or is at least pretty hard to reproduce with current tip, so let's push it down to the next milestone.

**Comment by tpgxyz** [comment]
I cant reporoduce this issue with systemd-248 anymore. Thanks for fixing it.

**Comment by mikhailnov** [comment]
I and a colleague of mine are also hit by this on 246.12, is help with debugging still needed?

**Comment by subpop** [comment]
I'm experiencing this same error with a DNS record provided by DNS on a VPN link on Fedora 34 (v248). I'm happy to provide logs of systemd-resolved off-list. I'm not sure about the level of privacy I should maintain with respect to the domain names from this DNS service (my employer).

I'm getting this error message though (domain name redacted):
```
Jun 17 14:45:24 subpop systemd-resolved[12699]: Sent message type=error sender=n/a destination=:1.435 path=n/a interface=n/a member=n/a cookie=42 reply_cookie=2 signature=s error-name=org.freedesktop.resolve1.NoSuchRR error-message='<domain-name>' does not have any RR of the requested type
```

**Comment by MartinX3** [comment]
Same here with my own dns entry from freedns.afraid.org
Using systemd 249

**Comment by cstrahan-blueshift** [comment]
I'm believe I'm also hitting this:

```
$ resolvectl query REDACTED.us-west-2.elb.amazonaws.com
REDACTED.us-west-2.elb.amazonaws.com: resolve call failed: 'REDACTED.us-west-2.elb.amazonaws.com' does not have any RR of the requested type
```

In case it helps:

```
$ apt list --installed | grep systemd

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

libnss-systemd/jammy,now 249.11-0ubuntu3pop0~1663192488~22.04~d6043af amd64 [installed,automatic]
libpam-systemd/jammy,now 249.11-0ubuntu3pop0~1663192488~22.04~d6043af amd64 [installed,automatic]
libsystemd0/jammy,now 249.11-0ubuntu3pop0~1663192488~22.04~d6043af amd64 [installed,automatic]
python3-systemd/jammy,now 234-3ubuntu2 amd64 [installed,automatic]
systemd-sysv/jammy,now 249.11-0ubuntu3pop0~1663192488~22.04~d6043af amd64 [installed,automatic]
systemd-timesyncd/jammy,now 249.11-0ubuntu3pop0~1663192488~22.04~d6043af amd64 [installed,automatic]
systemd/jammy,now 249.11-0ubuntu3pop0~1663192488~22.04~d6043af amd64 [installed,automatic]

$ cat /etc/os-release
NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os

$ uname -a
Linux 8l05sc182305 5.19.16-76051916-generic #202210150742~1666053244~22.04~cf07008 SMP PREEMPT_DYNAMIC Tue O x86_64 x86_64 x86_64 GNU/Linux
```
